### PR TITLE
STORM-2909 port new metrics to 2.x branch

### PR DESCRIPTION
**Pull request by agresch**
@ptgoetz 

Ported the new metrics code from 1.2 to 2.x.   Verified ConsoleReporter spit out some of the new metrics.

### CODE REVIEW

File: storm-client/src/jvm/org/apache/storm/daemon/worker/Worker.java [hunk 1]
```diff
@@ -223,7 +224,7 @@ private Object loadWorker(Map<String, Object> topologyConf, IStateStorage stateS
 
 
         // This thread will send out messages destined for remote tasks (on other workers)
-        if ( ( (Long)topologyConf.get(Config.TOPOLOGY_WORKERS) ) > 1 ) {
+        if (((Long)topologyConf.get(Config.TOPOLOGY_WORKERS)) > 1) {
```
Comment (HeartSaVioR) [inline storm-client/src/jvm/org/apache/storm/daemon/worker/Worker.java]: Please rebase with latest master.

File: storm-server/src/main/java/org/apache/storm/nimbus/DefaultTopologyValidator.java [hunk 2]
```diff
@@ -15,19 +15,53 @@
...
-import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class DefaultTopologyValidator implements ITopologyValidator {
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultTopologyValidator.class);
 
     @Override
-    public void prepare(Map<String, Object> StormConf){
+    public void prepare(Map stormConf){
```
Comment (HeartSaVioR) [inline storm-server/src/main/java/org/apache/storm/nimbus/DefaultTopologyValidator.java]: Please keep exposing generic type if possible.

**Comment by HeartSaVioR** [comment]
Could you take up copying documentation from another PR as well, based on #2526?

**Comment by agresch** [comment]
@HeartSaVioR  made the changes you suggested.  I can look at the documentation changes.

**Pull request merged**